tests/fsharp/Compiler/CodeGen/EmittedIL/CeEdiThrow.fs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+namespace FSharp.Compiler.UnitTests.CodeGen.EmittedIL
+
+open FSharp.Compiler.UnitTests
+open NUnit.Framework
+
+#if NETCOREAPP
+[<TestFixture>]
+module CeEdiThrow =
+
+    [<Test>]
+    let ``Emits EDI.Throw``() =
+        CompilerAssert.CompileLibraryAndVerifyIL
+            """
+module CE
+
+open System
+type Try() =
+    member _.Return _ = ()
+    member _.Delay f = f
+    member _.Run f = f()
+    member _.TryWith(body : unit -> unit, catch : exn -> unit) =
+        try body() with ex -> catch ex
+
+let foo = Try(){
+    try return invalidOp "Ex"
+    with :? ArgumentException -> return ()
+}
+            """
+            (fun verifier -> verifier.VerifyIL [
+            """
+.method public strict virtual instance class [FSharp.Core]Microsoft.FSharp.Core.Unit
+        Invoke(class [runtime]System.Exception _arg1) cil managed
+{
+
+  .maxstack  5
+  .locals init (class [runtime]System.ArgumentException V_0)
+  IL_0000:  ldarg.1
+  IL_0001:  isinst     [runtime]System.ArgumentException
+  IL_0006:  stloc.0
+  IL_0007:  ldloc.0
+  IL_0008:  brfalse.s  IL_000c
+
+  IL_000a:  ldnull
+  IL_000b:  ret
+
+  IL_000c:  ldarg.1
+  IL_000d:  call       void [runtime]System.Runtime.ExceptionServices.ExceptionDispatchInfo::Throw(class [runtime]System.Exception)
+  IL_0012:  ldnull
```

Comment:
This ok? It'll never be reached but the emitted cast will be unsound (in this case the CE is returning unit, for other types `IL_000a` will be correct while IL_0012 will stay `ldnull`)

Reply:
does it pass PEVerify?

I did a similar test, and it failed:
![image](https://user-images.githubusercontent.com/4236651/78513103-93bc7000-77a9-11ea-9da6-1bdc93d197ec.png)

> il-test.exe : il_test.Program::M][offset 0x00000001][found Nullobjref 'NullReference'][expected Int32] Unexpected type on the stack.

Reply:
@0x53A thanks, that's helpful, it won't be truly on the stack though, EDI.Throw never returns. I guess PEVerify isn't available in unix flavor is it?

Reply:
Not related to this issue per say, but it's worth noting that PEVerify and recent C#/.NET constructs don't always get along. Example: https://github.com/dotnet/roslyn/issues/27382

Reply:
See comment above about adding a `mkSequential`

Reply:
Fixed, emitting default now